### PR TITLE
Search options in new editor

### PR DESCRIPTION
**Pull request by pskowronek**
initial commit for review, will continue to add Replace - #879

### CODE REVIEW

**Comment by pskowronek** [comment]
I don't know why `jre` stuck to my commit (due to `rebase` somehow).

File: mucommander-core/src/main/java/com/mucommander/search/SearchBuilder.java [hunk 1]
```diff
@@ -93,19 +93,11 @@ public SearchBuilder where(AbstractFile entrypoint) {
         return this;
     }
 
-    public SearchBuilder depth(int searchDepth) {
+    public SearchBuilder depth(int searchDepth) { // TODO not used?
```
Comment (ahadas) [inline mucommander-core/src/main/java/com/mucommander/search/SearchBuilder.java:96]: yeah, seems it was replaced with `searchDepth`

File: mucommander-core/src/main/java/com/mucommander/search/LastSearchQuery.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.mucommander.search;
+
+/**
+ * A class to keep the last search query and its options.
+ */
+public final class LastSearchQuery {
```
Comment (ahadas) [inline mucommander-core/src/main/java/com/mucommander/search/LastSearchQuery.java:22]: why do we need this class, can't we use the values from SearchProperty instead?

File: mucommander-core/src/main/java/com/mucommander/search/SearchDialog.java [hunk 3]
```diff
@@ -241,6 +238,7 @@ private void cleanError() {
         mainPanel.addSpace(10);
 
         YBoxPanel textSearchPanel = new YBoxPanel(10);
+        // TODO FIXME "Text search (Optional)" - not in dictionary?
```
Comment (ahadas) [inline mucommander-core/src/main/java/com/mucommander/search/SearchDialog.java:241]: yeah, should be in the dictionary

File: jre
Comment (ahadas) [inline jre:1]: is this change intentional?

File: mucommander-translator/src/main/resources/dictionary.properties [hunk 4]
```diff
@@ -601,6 +601,13 @@ text_viewer.tabs_emulated = Tabs emulated with spaces
 text_viewer.tab_lines = Tab lines
 text_viewer.tab_size = Tab size
 text_viewer.whitespace_visible = Whitespace visible
+text_viewer.find_replace.backward = Backward
+text_viewer.find_replace.case_sensitive = Case sensitive
```
Comment (ahadas) [inline mucommander-translator/src/main/resources/dictionary.properties:605]: we can point to another occurrence of 'case sensitive'

File: mucommander-translator/src/main/resources/dictionary.properties [hunk 5]
```diff
@@ -601,6 +601,13 @@ text_viewer.tabs_emulated = Tabs emulated with spaces
...
+text_viewer.find_replace.direction = Direction
+text_viewer.find_replace.forward = Forward
+text_viewer.find_replace.options = Options
+text_viewer.find_replace.regex_match = Regular expressions
```
Comment (ahadas) [inline mucommander-translator/src/main/resources/dictionary.properties:609]: also here, better to point to an existing entry

File: jre
Comment (ahadas) [reply jre:1]: ah ok, I see your comment about that

File: mucommander-viewer-text/build.gradle [hunk 6]
```diff
@@ -23,7 +23,7 @@ jar {
         'Bundle-Description': 'Library for textual viewer/editor',
         'Bundle-DocURL': 'https://www.mucommander.com',
         'Export-Package': 'com.mucommander.viewer.text',
-        'Import-Package': 'org.fife.ui.rsyntaxtextarea;resolution:=dynamic,*',
+        'Import-Package': 'com.mucommander.ui.text;resolution:=dynamic,com.mucommander.search;resolution:=dynamic,org.fife.ui.rsyntaxtextarea;resolution:=dynamic,*',
```
Comment (ahadas) [inline mucommander-viewer-text/build.gradle:26]: why is this needed when the packages are exported?

File: mucommander-core/src/main/java/com/mucommander/search/LastSearchQuery.java (hunk 2 above)
Comment (ahadas) [reply mucommander-core/src/main/java/com/mucommander/search/LastSearchQuery.java:22]: posted #903 with the alternative I had in mind

File: mucommander-viewer-text/build.gradle (hunk 6 above)
Comment (ahadas) [reply mucommander-viewer-text/build.gradle:26]: also covered in #903 (I meant the `resolution:=dynamic` part)

File: mucommander-core/src/main/java/com/mucommander/search/LastSearchQuery.java (hunk 2 above)
Comment (pskowronek) [reply mucommander-core/src/main/java/com/mucommander/search/LastSearchQuery.java:22]: Not sure if I like representing `boolean` values as strings.... but OK, I will follow your idea.

**Comment by pskowronek** [comment]
PR updated with all remarks addressed I think. If it is OK then it can be merged, then I will continue with Replace functionality.

@ahadas any hint how to know inside `TextEditor` if file was opened from search results? I could then run `Find` to move the first occurrence of searched string.

File: mucommander-translator/src/main/resources/dictionary_en.properties [hunk 7]
```diff
@@ -484,7 +484,8 @@ search_dialog.max_threads = Maximum
 search_dialog.case_sensitive = $[file_selection_dialog.case_sensitive]
 search_dialog.matches_regexp = $[file_selection_dialog.matches_regexp]
 search_dialog.wildcards = (* = any string, ? = any character, \\ = escape for literals: *?)
-search_dialog.search_text  = Containing text
+search_dialog.search_text = Containing text
+search_dialog.search_text_optional = Text search (Optional)
```
Comment (ahadas) [inline mucommander-translator/src/main/resources/dictionary_en.properties:488]: why here and not in dictionary.properties?

**Review by ahadas — APPROVED** [review]

**Comment by ahadas** [comment]
just one minor comment above

> @ahadas any hint how to know inside `TextEditor` if file was opened from search results? I could then run `Find` to move to the first occurrence of searched string.

that would be nice indeed. I don't see a "clean" way to do this but to take the location that is presented in the active folder panel and check if it starts with a search-scheme

File: mucommander-translator/src/main/resources/dictionary_en.properties (hunk 7 above)
Comment (pskowronek) [reply mucommander-translator/src/main/resources/dictionary_en.properties]: right, I've updated PR

**Review by ahadas — APPROVED** [review]
